### PR TITLE
Handle provider meta better in PlanResourceChange

### PR DESCRIPTION
**Pull request by VenelinMartinov**
fixes https://github.com/pulumi/pulumi-terraform-bridge/issues/1822 and unlocks https://github.com/pulumi/pulumi-gcp/issues/1874

`ProviderMeta` seems to be an old experimental feature of TF which has not picked up.

AWS does not use it and GCP uses it only for one thing - setting UserAgent strings: https://github.com/modular-magician/terraform-provider-google-beta/commit/6cf6c51b6b3804d6ce050885c98ae030f378bd7e

The meta we receive does not match the schema provided and does not contain the module_name property. We should set it to null and potentially revisit if this causes issues. I've tested it for the GCP resource in https://github.com/pulumi/pulumi-terraform-bridge/pull/1825 and works fine.

I've opened https://github.com/pulumi/pulumi-terraform-bridge/issues/1827 as a follow-up to test with one of the resources which actually use the ProviderMeta.

### CODE REVIEW

**Comment by iwahbe** [comment]
I don't really understand this change. I think a test of would make it clearer for me.

**Comment by VenelinMartinov** [comment]
This code would panic on the GCP provider. It is not used in AWS and nothing else uses the PlanResourceChange stuff. 

I'd like to use PlanResourceChange in GCP. To do that, I've "fixed" the `providerMeta` to always return null.

There's a regression test in https://github.com/pulumi/pulumi-terraform-bridge/pull/1825/files but it's failing due to unrelated reasons.

File: pkg/tfshim/sdk-v2/provider2.go [hunk 1]
```diff
@@ -282,15 +282,7 @@ func (p *planResourceChangeImpl) providerMeta() (*cty.Value, error) {
...
-	m := p.tf.Meta()
-	if m == nil {
-		v := cty.NullVal(metaSchema.ImpliedType())
-		return &v, nil
-	}
-	v, err := recoverAndCoerceCtyValueWithSchema(metaSchema, m)
-	if err != nil {
-		return nil, fmt.Errorf("ProviderMeta: %w", err)
-	}
+	v := cty.NullVal(metaSchema.ImpliedType())
```
Comment (t0yv0) [inline pkg/tfshim/sdk-v2/provider2.go:285]: Add a test please. Otherwise LGTM. Some of your PR comment may be worth repeating in source.

**Review by t0yv0 — APPROVED** [review]

**Comment by t0yv0** [comment]
I think p.tf.Meta() and ProvderMetaSchema are just different things right? I remember bumping into this before, it keeps being difficult to remember because they're both named Meta. I must have crossed the wrong wires when writing providerv2 code.

**Comment by iwahbe** [comment]
> I think p.tf.Meta() and ProvderMetaSchema are just different things right? I remember bumping into this before, it keeps being difficult to remember because they're both named Meta. I must have crossed the wrong wires when writing providerv2 code.

I'm pretty sure that they are unrelated. `p.tf.Meta()` is:

```go
// Meta returns the metadata associated with this provider that was
// returned by the Configure call. It will be nil until Configure is called.
func (p *Provider) Meta() interface{} {
	return p.meta
}
```

I'm pretty sure that it is provider local and never needs to be serialized. I think `ProvderMetaSchema` is the schema (at least in theory) for the config that was passed into the provider (the arguments for `Configure`). @t0yv0 I think the problem is you have the input and output domains confused in `func (p *planResourceChangeImpl) providerMeta() (*cty.Value, error)`.

File: pkg/tfshim/sdk-v2/provider2.go [hunk 2]
```diff
@@ -278,20 +278,8 @@ func (p *planResourceChangeImpl) Importer(t string) shim.ImportFunc {
...
-		v := cty.NullVal(metaSchema.ImpliedType())
-		return &v, nil
-	}
-	v, err := recoverAndCoerceCtyValueWithSchema(metaSchema, m)
-	if err != nil {
-		return nil, fmt.Errorf("ProviderMeta: %w", err)
-	}
-	return &v, nil
+	return nil, nil
+	// TODO: Implement this if needed. https://github.com/pulumi/pulumi-terraform-bridge/issues/1827
```
Comment (iwahbe) [inline pkg/tfshim/sdk-v2/provider2.go:282]: From our engineering process doc: 

```
Added TODOs must include links to issue numbers in TODO[pulumi/pulumi#7973] format.
```

```suggestion
	// TODO[pulumi/pulumi-terraform-bridge#1827]: We do not believe that this is load bearing in any providers.
```

**Review by iwahbe — APPROVED** [review]

**Pull request merged**